### PR TITLE
[SPARK-46121][PYTHON][DOCS] Refine docstring of `concat/array_position/element_at/try_element_at`

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr refine docstring of `concat/array_position/element_at/try_element_at` and add some new examples.

### Why are the changes needed?
To improve PySpark documentation

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass Github Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -379,6 +379,7 @@ jobs:
       SKIP_PACKAGING: true
       METASPACE_SIZE: 1g
       BRANCH: ${{ inputs.branch }}
+      SPARK_ANSI_SQL_MODE: true
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:382]: Check ansi is true , will revert this file after the test passes.

Comment (LuciferYang) [reply .github/workflows/build_and_test.yml:382]: https://github.com/LuciferYang/spark/runs/19046429659

image


SPARK_ANSI_SQL_MODE check passed

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
Thanks @HyukjinKwon ~

**Pull request closed without merging**